zio/src/main/scala/sttp/capabilities/zio/ZioStreams.scala

```diff
@@ -1,10 +1,49 @@
 package sttp.capabilities.zio
 
+import sttp.capabilities.StreamMaxLengthExceeded
 import sttp.capabilities.Streams
+import zio.Chunk
+import zio.Trace
 import zio.stream.Stream
+import zio.stream.ZChannel
+import zio.stream.ZStream
+
+import scala.util.control.NonFatal
 
 trait ZioStreams extends Streams[ZioStreams] {
   override type BinaryStream = Stream[Throwable, Byte]
   override type Pipe[A, B] = Stream[Throwable, A] => Stream[Throwable, B]
 }
-object ZioStreams extends ZioStreams
+
+object ZioStreams extends ZioStreams {
+
+  def limitBytes(stream: Stream[Throwable, Byte], maxBytes: Long): Stream[Throwable, Byte] =
+    scanChunksAccum(stream, initState = 0L) { case (accumulatedBytes, chunk) =>
+      val byteCount = accumulatedBytes + chunk.size
+      if (byteCount > maxBytes)
+        throw new StreamMaxLengthExceeded(maxBytes)
+      else
+        byteCount
+    }
+
+  def scanChunksAccum[S, R, A](inputStream: ZStream[R, Throwable, A], initState: => S)(
```

Comment:
make this private?

Reply:
I thought it might be harmless when exposed, because it's a general utility function to process any stream's underlying chunks.

Reply:
the smaller the public API, the better :) I think you had some experiences already with MiMA ;)